### PR TITLE
add plant picture in firebase

### PR DESCRIPTION
**Pull request by Lonexera**

### CODE REVIEW

File: app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebasePlantsRepository.kt [hunk 1]
```diff
@@ -30,10 +42,16 @@ class FirebasePlantsRepository @Inject constructor(
                         }
 
                         value != null -> {
-                            val plants = value
-                                .toObjects<FirebasePlant>()
-                                .map { it.toPlant() }
-                            trySend(plants)
+                            launch {
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebasePlantsRepository.kt:45]: Why you started one more launch?

File: app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebasePlantsRepository.kt [hunk 2]
```diff
@@ -49,13 +67,20 @@ class FirebasePlantsRepository @Inject constructor(
             .get()
             .await()
             .toObjects<FirebasePlant>()
-            .map { it.toPlant() }
+            .map {
+                it.toPlant(
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebasePlantsRepository.kt:71]: Is in necessary to save image to local storage? 
Maybe you can just use url?

https://firebase.google.com/docs/storage/android/upload-files#kotlin+ktx

image

File: app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebasePlantsRepository.kt [hunk 3]
```diff
@@ -77,8 +106,51 @@ class FirebasePlantsRepository @Inject constructor(
...
+
+                if (!localFile.exists()) {
+                    imageRef.getFile(localFile).await()
+                }
+
+                context.getFileUri(localFile)
+            }
+            else -> null
+        }
+    }
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebasePlantsRepository.kt:137]: Have a question about it: is it nesessary to do so? Or maybe we can just use some url?

File: app/src/main/java/com/example/plantsapp/data/firebase/entity/FirebasePlant.kt [hunk 4]
```diff
@@ -9,20 +9,20 @@ data class FirebasePlant(
     val plantPicture: String = ""
 ) {
 
-    fun toPlant(): Plant {
+    fun toPlant(localImageUri: Uri?): Plant {
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/data/firebase/entity/FirebasePlant.kt:12]: Actually - probably you can store image , get it id and in FirebasePlant store link to this image as plantPicture plant

**Review by Vladuken — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebasePlantsRepository.kt (hunk 1 above)
Comment (Lonexera) [reply app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebasePlantsRepository.kt:45]: needed it here for accessing suspend function to save image locally

**Pull request merged**